### PR TITLE
cmd/snap-preseed: reset run inhibit locks on --reset.

### PR DESCRIPTION
**Pull request by stolowski**
Reset inhibit locks on `snap-preseed --reset`.

I suppose we could avoid creating locks when preseeding, but I'm not sure we need to special-case there.

This should fix snap-preseed spread test failure on master.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thank you

File: cmd/snap-preseed/reset.go [hunk 1]
```diff
@@ -60,6 +61,7 @@ func resetPreseededChroot(preseedChroot string) error {
 		filepath.Join(dirs.SnapUserServicesDir, "default.target.wants", "snap.*.service"),
 		filepath.Join(dirs.SnapUserServicesDir, "sockets.target.wants", "snap.*.socket"),
 		filepath.Join(dirs.SnapUserServicesDir, "timers.target.wants", "snap.*.timer"),
+		filepath.Join(runinhibit.InhibitDir, "*.lock"),
```
Comment (bboozzoo) [inline cmd/snap-preseed/reset.go:64]: Should this glob be more specific and just use `snapd.lock` ?

File: cmd/snap-preseed/main_test.go [hunk 2]
```diff
@@ -616,6 +617,7 @@ func (s *startPreseedSuite) TestReset(c *C) {
 			{filepath.Join(dirs.SnapSeqDir, "foo.json"), ""},
 			{filepath.Join(dirs.SnapMountDir, "foo", "bin"), ""},
 			{filepath.Join(dirs.SnapSeccompDir, "foo.bin"), ""},
+			{filepath.Join(runinhibit.InhibitDir, "snapd.lock"), ""},
```
Comment (pedronis) [inline cmd/snap-preseed/main_test.go:620]: this should probably be foo.lock for clarity

File: cmd/snap-preseed/reset.go (hunk 1 above)
Comment (pedronis) [reply cmd/snap-preseed/reset.go:64]: no, there are many locks there called \<instance-name>.lock.  The test is just a bit misleading right now I think

**Review by pedronis — APPROVED** [review]
thanks, but see comments about test clarity

**Review by anonymouse64 — APPROVED** [review]
LGTM, thanks for this

**Comment by anonymouse64** [comment]
@mvo5 I milestoned this for 2.48, as it is currently causing required spread tests to fail, so we will need this if we want green spread tests on 2.48 branch

**Pull request merged**